tools/mkconfig.c

```diff
@@ -84,64 +84,15 @@ int main(int argc, char **argv, char **envp)
     "/* config.h -- Autogenerated! Do not edit. */\n\n"
     "#ifndef __INCLUDE_NUTTX_CONFIG_H\n"
     "#define __INCLUDE_NUTTX_CONFIG_H\n\n"
-    "/* General Definitions ***********************************/\n"
     "/* Used to represent the values of tristate options */\n\n"
     "#define CONFIG_y 1\n"
     "#define CONFIG_m 2\n\n"
-    "/* Architecture-specific options *************************/\n\n");
+    "/* General Definitions ***********************************/\n");
 
   generate_definitions(stream);
 
   printf(
     "\n/* Sanity Checks *****************************************/\n\n"
-    "/* If this is an NXFLAT, external build, then make sure that\n"
-    " * NXFLAT support is enabled in the base code.\n"
-    " */\n\n"
-    "#if defined(__NXFLAT__) && !defined(CONFIG_NXFLAT)\n"
-    "# error \"NXFLAT support not enabled in this configuration\"\n"
-    "#endif\n\n"
-    "/* NXFLAT requires PIC support in the TCBs. */\n\n"
-    "#if defined(CONFIG_NXFLAT)\n"
-    "#  undef  CONFIG_PIC\n"
-    "#  define CONFIG_PIC 1\n"
-    "#endif\n\n"
-    "/* Binary format support is disabled if no binary formats are\n"
-    " * configured (at present, NXFLAT is the only supported binary.\n"
-    " * format).\n"
-    " */\n\n"
-    "#if !defined(CONFIG_NXFLAT) && !defined(CONFIG_ELF) \\\n"
-    "    && !defined(CONFIG_BUILTIN)\n"
-    "#  undef CONFIG_BINFMT_DISABLE\n"
-    "#  define CONFIG_BINFMT_DISABLE 1\n"
-    "#endif\n\n"
-    "/* The correct way to disable RR scheduling is to set the\n"
-    " * timeslice to zero.\n"
-    " */\n\n"
-    "#ifndef CONFIG_RR_INTERVAL\n"
-    "#  define CONFIG_RR_INTERVAL 0\n"
-    "#endif\n\n"
-    "/* If no file streams are configured, then make certain that\n"
-    " * buffered I/O support is disabled\n"
-    " */\n\n"
-    "#ifndef CONFIG_FILE_STREAM\n"
-    "#  undef  CONFIG_STDIO_BUFFER_SIZE\n"
-    "#  undef  CONFIG_STDIO_LINEBUFFER\n"
-    "#  undef  CONFIG_STDIO_DISABLE_BUFFERING\n"
-    "#  define CONFIG_STDIO_DISABLE_BUFFERING 1\n"
-    "#endif\n\n"
-    "/* If priority inheritance is disabled, then do not allocate any\n"
-    " * associated resources.\n"
-    " */\n\n"
-    "#if !defined(CONFIG_PRIORITY_INHERITANCE) \\\n"
-    "    || !defined(CONFIG_SEM_PREALLOCHOLDERS)\n"
-    "#  undef CONFIG_SEM_PREALLOCHOLDERS\n"
-    "#  define CONFIG_SEM_PREALLOCHOLDERS 0\n"
-    "#endif\n\n"
-    "#if !defined(CONFIG_PRIORITY_INHERITANCE) \\\n"
-    "    || !defined(CONFIG_SEM_NNESTPRIO)\n"
-    "#  undef  CONFIG_SEM_NNESTPRIO\n"
-    "#  define CONFIG_SEM_NNESTPRIO 0\n"
-    "#endif\n\n"
```

Comment:
Why to remove this explanation? Isn't valid anymore? Could it be fixed to reflect the new PRIORITY_INHERITANCE modification and maybe it could become a comment at the source file.

Reply:
If the dependence between Kconfig exist, it's better to add them explicitly in the related header/source files, instead inserting to the generated header files.

> Why to remove this explanation? 

What's explanation? I just want to remove the define/undef.

Isn't valid anymore?

If it's still valid, they should put into the related header/source files just like many other Kconfig. I don't think these Kconfig is enough special to treat them difference.

> Could it be fixed to reflect the new PRIORITY_INHERITANCE modification and maybe it could become a comment at the source file.